src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,231 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableResult;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.plugin.utils.SalesforceClient;
+import org.junit.Assert;
+
+import java.io.IOException;
+import java.sql.Date;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+
+  static List<JsonObject> bigQueryResponse = new ArrayList<>();
+  static List<Object> bigQueryRows = new ArrayList<>();
+  static Gson gson = new Gson();
+
+  public static boolean validateSalesforceToBQRecordValues(String objectName, String targetTable) throws
+    IOException, InterruptedException {
+    String id = SalesforceClient.id;
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateBQToSalesforceRecordValues(String targetTable, String objectName) throws
+    IOException, InterruptedException {
+    SalesforceClient.queryObjectID(objectName);
+    String id = SalesforceClient.id;
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateSalesforceMultiObjectToBQ(String objectName) throws IOException,
```

Comment:
Update this logic as it won't work for 2 different Sobjects

Reply:
done